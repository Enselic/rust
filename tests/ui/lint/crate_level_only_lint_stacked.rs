//! Tests that if mulitple crate_level_only lints are active at the same time,
//! both of them are reported as errors, but only one time each.

#![deny(unused_attributes)]
#![allow(dead_code)]

#[allow(uncommon_codepoints)]
//~^ ERROR allow(uncommon_codepoints) is ignored unless specified at crate level [unused_attributes]
fn main() {
    #[allow(non_ascii_idents)]
    //~^ ERROR allow(non_ascii_idents) is ignored unless specified at crate level [unused_attributes]
    //~| ERROR allow(non_ascii_idents) is ignored unless specified at crate level [unused_attributes]
    const ÅÅÅÅ: f64 = 0.000001;
}
