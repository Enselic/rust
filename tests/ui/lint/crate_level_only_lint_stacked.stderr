error: allow(uncommon_codepoints) is ignored unless specified at crate level
  --> $DIR/crate_level_only_lint_stacked.rs:7:9
   |
LL | #[allow(uncommon_codepoints)]
   |         ^^^^^^^^^^^^^^^^^^^
   |
note: the lint level is defined here
  --> $DIR/crate_level_only_lint_stacked.rs:4:9
   |
LL | #![deny(unused_attributes)]
   |         ^^^^^^^^^^^^^^^^^

error: allow(non_ascii_idents) is ignored unless specified at crate level
  --> $DIR/crate_level_only_lint_stacked.rs:10:13
   |
LL |     #[allow(non_ascii_idents)]
   |             ^^^^^^^^^^^^^^^^

error: allow(non_ascii_idents) is ignored unless specified at crate level
  --> $DIR/crate_level_only_lint_stacked.rs:10:13
   |
LL |     #[allow(non_ascii_idents)]
   |             ^^^^^^^^^^^^^^^^
   |
   = note: duplicate diagnostic emitted due to `-Z deduplicate-diagnostics=no`

error: aborting due to 3 previous errors

